spec/models/reservation_spec.rb

```diff
@@ -21,6 +21,43 @@
   # can't run if nil (?)
   #
 
+  describe '.number_for' do
+    before(:each) do
+      @source = FactoryGirl.build_pair(:valid_reservation)
+    end
+    it 'counts the number that overlap with today' do
+      expect(Reservation.number_for(@source)).to eq(2)
+    end
+    it 'counts the number that overlap with a given day' do
+      @source.last.assign_attributes(start_date: Time.zone.today + 2,
+                                     due_date: Time.zone.today + 3)
+      expect(Reservation.number_for(@source, date: Time.zone.today + 2.days)
+            ).to eq(1)
+    end
+    it 'can count only not overdue reservations' do
+      @source.last.attributes = FactoryGirl.attributes_for(
```

Comment:
This is cool (the `attributes=` method); intuitively it felt like this would overwrite the other attributes but I see that it's just an alias for `assign_attributes`. Nice find!